server/src/main/java/org/elasticsearch/index/fielddata/ScriptDocValues.java

```diff
@@ -177,15 +206,27 @@ public int size() {
         private static final ReadableDateTime EPOCH = new DateTime(0, DateTimeZone.UTC);
 
         private final SortedNumericDocValues in;
+        private final Consumer<String> deprecationCallback;
```

Comment:
Why do we need this abstracted? Can't the Longs.deprecated method just call deprecationLogger.deprecated directly?

Reply:
It could but then I couldn't test this. There isn't any consistent way to make it fail without this funny unit test dance that I do because it usually only fails when you rotate the log files. At least, that is the only time I could get it to fail locally.